manager/apis/app/v1alpha1/fybrikapplication_types.go

```diff
@@ -9,26 +9,29 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// CatalogRequirements contain the specifics for catalogging the data asset
-type CatalogRequirements struct {
-	// CatalogService specifies the datacatalog service that will be used for catalogging the data into.
-	// +optional
-	CatalogService string `json:"service,omitempty"`
-
+// TargetCatalog contain the specifics for the catalog which needs to be updated.
+// Updates may be adding a new dataset to the catalog (currently supported) or deleting an existing dataset (future)
+type TargetCatalog struct {
 	// CatalogID specifies the catalog where the data will be cataloged.
 	// +optional
 	CatalogID string `json:"catalogID,omitempty"`
 }
 
-// CopyRequirements include the requirements for the data copy operation
-type CopyRequirements struct {
-	// Required indicates that the data must be copied.
+// FlowRequirements include the requirements for the data copy operation when there is an explicit copy request
+// Note: Implicit copies done for data plane optimization by Fybrik do not use these parameters
+type FlowRequirements struct {
+	// Catalog indicates that the data asset must be cataloged, and in which catalog to register it.
 	// +optional
-	Required bool `json:"required,omitempty"`
+	Catalog TargetCatalog `json:"catalog,omitempty"`
 
-	// Catalog indicates that the data asset must be cataloged.
+	// Storage estimate indicates the estimated amount of storage required when copying or writing new data.
 	// +optional
-	Catalog CatalogRequirements `json:"catalog,omitempty"`
+	StorageEstimate int `json:"storageEstimate"`
+
+	// IsNewDataSet if true indicates that the DataContext.DataSetID is user provided and not a true catalog ID.  Relevant when writing.
+	// A unique ID from the catalog will be provided in the FybrikApplication Status after a new catalog entry is created.
+	// +optional
+	IsNewDataSet bool `json:"isNewDataSet"`
```

Comment:
So maybe allow an empty datasetID instead if there is no dataset?

Reply:
See answer to following comment